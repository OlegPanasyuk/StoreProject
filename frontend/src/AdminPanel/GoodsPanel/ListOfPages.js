import React, { Component } from 'react';
import PropTypes from 'prop-types';

//Components
import { Nav, Pagination } from 'react-bootstrap';

export class ListOfPages extends Component {
    render() {
        let { 
            count, 
            limit, 
            activePage, 
            openPage
        } = this.props;
        const numberOfPages = Math.ceil(count / limit);
        let listOfPages = [];
        for (let i = 1; i <= numberOfPages; i++) {
            listOfPages.push((
                <Nav.Item 
                    as={Pagination.Item} 
                    key={`nav-page-list-${i}`} 
                    active={i === activePage} 
                    onClick={() => openPage(i)}
                >
                    {i}
                </Nav.Item>
            ));
        }
        return (
            <Nav as={Pagination} size='sm' className='d-flex justify-content-center' >
                {listOfPages}
            </Nav>
        );
    }
}

ListOfPages.propTypes = {
    count: PropTypes.number,
    limit: PropTypes.number,
    activePage: PropTypes.number,
    openPage: PropTypes.func
};

export default ListOfPages;