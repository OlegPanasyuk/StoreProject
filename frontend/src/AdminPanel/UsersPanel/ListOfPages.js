import React, { Component } from 'react';

//Components
import { Nav, Pagination } from 'react-bootstrap';

export class ListOfPages extends Component {
    render() {
        let { count, limit, activePage, openPage } = this.props;
        const numberOfPages = Math.ceil(count / limit);
        let listOfPages = [];
        for (let i = 1; i <= numberOfPages; i++) {
            listOfPages.push((
                <Nav.Item 
                    as={Pagination.Item} 
                    key={`nav-page-list-${i}`} 
                    active={i === activePage} 
                    onClick={() => openPage(i)}
                >
                    {i}
                </Nav.Item>
            ));
        }
        return (
            <Nav as={Pagination} size='sm' className='d-flex justify-content-center' >
                {listOfPages}
            </Nav>
        );
    }
}

export default ListOfPages;