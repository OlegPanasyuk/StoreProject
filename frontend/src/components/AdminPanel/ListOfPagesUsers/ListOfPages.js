import React from 'react';
import PropTypes from 'prop-types';

// Components
import { Nav, Pagination } from 'react-bootstrap';

function ListOfPages({
    count,
    limit,
    activePage,
    openPage
}) {
    const numberOfPages = Math.ceil(count / limit);
    const listOfPages = [];
    for (let i = 1; i <= numberOfPages; i++) {
        listOfPages.push((
            <Nav.Item
                as={Pagination.Item}
                key={`nav-page-list-${i}`}
                active={i === activePage}
                onClick={() => openPage(i)}
            >
                {i}
            </Nav.Item>
        ));
    }
    return (
        <Nav as={Pagination} size='sm' className='d-flex justify-content-center'>
            {listOfPages}
        </Nav>
    );
}


ListOfPages.propTypes = {
    count: PropTypes.number,
    limit: PropTypes.number,
    activePage: PropTypes.number,
    openPage: PropTypes.func
};

ListOfPages.defaultProps = {
    count: 0,
    limit: 10,
    activePage: 1,
    openPage: () => null
};

export default ListOfPages;
