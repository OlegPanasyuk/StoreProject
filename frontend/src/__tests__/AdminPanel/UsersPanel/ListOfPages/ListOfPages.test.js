import React from 'react';
import ListOfPages from '../../../../AdminPanel/UsersPanel/ListOfPages';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    count: 108,
    limit: 10,
    activePage: 2,
    openPage: ()=>{}
}) {
    let component = shallow(<ListOfPages {...props} />);
    return { component, props };
}

describe('ListOfPages --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(ListOfPages.propTypes.count).toBe(PropTypes.number);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});
