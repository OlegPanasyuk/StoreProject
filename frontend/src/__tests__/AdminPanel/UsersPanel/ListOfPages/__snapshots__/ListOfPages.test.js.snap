// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOfPages --- Snapshot Component must be rendered 1`] = `
<Nav
  as={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "Ellipsis": [Function],
      "First": [Function],
      "Item": [Function],
      "Last": [Function],
      "Next": [Function],
      "Prev": [Function],
      "defaultProps": undefined,
      "displayName": "Bootstrap(Pagination)",
      "propTypes": undefined,
      "render": [Function],
    }
  }
  className="d-flex justify-content-center"
  fill={false}
  justify={false}
  size="sm"
>
  <NavItem
    active={false}
    as={[Function]}
    key="nav-page-list-1"
    onClick={[Function]}
  >
    1
  </NavItem>
  <NavItem
    active={true}
    as={[Function]}
    key="nav-page-list-2"
    onClick={[Function]}
  >
    2
  </NavItem>
  <NavItem
    active={false}
    as={[Function]}
    key="nav-page-list-3"
    onClick={[Function]}
  >
    3
  </NavItem>
  <NavItem
    active={false}
    as={[Function]}
    key="nav-page-list-4"
    onClick={[Function]}
  >
    4
  </NavItem>
  <NavItem
    active={false}
    as={[Function]}
    key="nav-page-list-5"
    onClick={[Function]}
  >
    5
  </NavItem>
  <NavItem
    active={false}
    as={[Function]}
    key="nav-page-list-6"
    onClick={[Function]}
  >
    6
  </NavItem>
  <NavItem
    active={false}
    as={[Function]}
    key="nav-page-list-7"
    onClick={[Function]}
  >
    7
  </NavItem>
  <NavItem
    active={false}
    as={[Function]}
    key="nav-page-list-8"
    onClick={[Function]}
  >
    8
  </NavItem>
  <NavItem
    active={false}
    as={[Function]}
    key="nav-page-list-9"
    onClick={[Function]}
  >
    9
  </NavItem>
  <NavItem
    active={false}
    as={[Function]}
    key="nav-page-list-10"
    onClick={[Function]}
  >
    10
  </NavItem>
  <NavItem
    active={false}
    as={[Function]}
    key="nav-page-list-11"
    onClick={[Function]}
  >
    11
  </NavItem>
</Nav>
`;
